content/2021-08-26-drg-trust/index.md

```diff
@@ -0,0 +1,202 @@
++++
+title = "Using X509 Client certificates for devices authentication with drogue-cloud"
+extra.author = "jb_and_vedang"
++++
+
+This article shows how you can setup X.509 certificates to authenticate devices connecting to Drogue Cloud and an example of how [`drg`](https://github.com/drogue-iot/drg) makes it easy.
+
+<!-- more -->
+
+# X509 Authentication?
+
+If you are already master how chains of trust works with X509 certs feel free to skip to the next part. Here is a quick explanation otherwise:
+
+Before dwelving into certificates stuff you need to know that they rely on *asymetric cryptography*. In two sentences, asymetric cryptography is a technique that allows encrytping/decrypting data with two keys.
+Let's take an key pair (A, B) : stuff encrypted with key A can only be decrypted with key B, and vice versa. In the certificates implementation of this technique, we name this key pair (Public, Private).
+
+You already use X509 and trust anchors in your daily life, right on this very blog ! It's the famous green padlock showing that you can trust that the connection is secured.
+But how could you trust that, if that is the first time you visit this website ?
+
+The security rely on a shared trusted entity. It's the same mechanism that you use when you accept bank notes from someone, you know the value is real because you trust the bank that created them.
```

Comment:
```suggestion
The security relies on a shared trusted entity. It's the same mechanism that you use when you accept bank notes from someone, you know the value is real because you trust the bank that created them.
```